### PR TITLE
(PLUGIN-539) Make schema names compatible with Avro naming

### PR DESCRIPTION
**Pull request by chtyim**

### CODE REVIEW

File: src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java [hunk 1]
```diff
@@ -241,9 +245,23 @@ public static Schema getSchemaWithFields(SObjectDescriptor sObjectDescriptor,
...
+  private static Schema createFieldSchema(Field field) {
     Schema fieldSchema = SALESFORCE_TYPE_TO_CDAP_SCHEMA.getOrDefault(field.getType(), DEFAULT_SCHEMA);
     return field.isNillable() ? Schema.nullableOf(fieldSchema) : fieldSchema;
   }
 
+  /**
+   * Normalize the given field name to be compatible with
+   * <a href="https://avro.apache.org/docs/current/spec.html#names">Avro names</a>
+   */
+  private static String normalizeAvroName(String name) {
```
Comment (fernst) [inline src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java:257]: nit: this method should have tests.

**Review by fernst — APPROVED** [review]
LGTM

File: src/test/java/io/cdap/plugin/salesforce/SalesforceSchemaUtilTest.java [hunk 2]
```diff
@@ -132,9 +132,12 @@ public void testGetSchemaWithFields() {
       Schema.Field.of("SumRelFunc", Schema.of(Schema.Type.LONG)),
       Schema.Field.of(contacts, Schema.arrayOf(Schema.recordOf(contacts,
         Schema.Field.of("FirstName", Schema.of(Schema.Type.STRING)),
-        Schema.Field.of("Owner.Status", Schema.of(Schema.Type.STRING))))));
+        Schema.Field.of("Owner_Status", Schema.of(Schema.Type.STRING))))));
 
-    Assert.assertEquals(expectedSchema.toString(), actualSchema.toString());
+    Assert.assertEquals(expectedSchema, actualSchema);
+
+    // Parse the schema as Avro schema to make sure all the naming are accepted by Avro
```
Comment (ssmith-google) [inline src/test/java/io/cdap/plugin/salesforce/SalesforceSchemaUtilTest.java]: Nit: "...make sure all the **names** are accepted by Avro"

Comment (chtyim) [reply src/test/java/io/cdap/plugin/salesforce/SalesforceSchemaUtilTest.java]: fixed

File: src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java (hunk 1 above)
Comment (chtyim) [reply src/main/java/io/cdap/plugin/salesforce/SalesforceSchemaUtil.java:257]: A avro parsing has been added to an existing test.

**Pull request merged**